src/pudl_archiver/depositors/zenodo.py

```diff
@@ -316,16 +318,29 @@ async def publish_deposition(self, deposition: Deposition) -> Deposition:
     async def delete_deposition(self, deposition) -> None:
         """Delete an un-submitted deposition.
 
+        As of 2023-11-22, Zenodo API times out on first few deletion attempts,
+        occasionally 500s, and then 404s once the delete has actually gone
+        through.
+
         Args:
             deposition: the deposition you want to delete.
         """
-        await self.request(
-            "DELETE",
-            deposition.links.self,
-            log_label="Deleting deposition",
-            headers=self.auth_write,
-            parse_json=False,
-        )
+        try:
+            await self.request(
+                "DELETE",
+                deposition.links.self,
+                log_label="Deleting deposition",
+                headers=self.auth_write,
+                parse_json=False,
+                retry_count=5,
+            )
+        except ZenodoClientError as e:
+            if e.status != 404:
+                raise e
+            logger.info(
+                f"Got 404 when deleting {deposition.links.self}, assuming "
```

Comment:
We could also look for 'persistent identifier does not exist' in the error message which I think would be a clearer determination that the deletion had occurred?

Reply:
Yeah, I think that's easier to read but also the HTTP status 404 is less likely to change in the future.

Reply:
Maybe just printing the error message would be helpful then, in case we see a very different 404 error message that suggests non-deletion and we want to revise our assumption.